test/carryPublicTokenCrowdsale.js

```diff
@@ -0,0 +1,228 @@
+        // See also presale constant on
+        // migrations/3_deploy_public_sale_contracts.js file.
+        fundWallet,  // wallet
+        token.address,  // token contract
+        65000,  // rate
+        web3.toWei(5000410, "finney"),  // cap
+        web3.toWei(99, "finney"),  // individualMinPurchaseWei
+        [
+            currentTimestamp + 14 * day,  // 2 weeks later
+            web3.toWei(5, "ether"),
+            currentTimestamp + 21 * day,  // 3 weeks later
+            web3.toWei(10, "ether"),
+            currentTimestamp + 31 * day,  // closingTime: a month later
+            0,
+        ],
+    ],
+    "CarryPublicTokenCrowdsale (opened; phase 1)": (fundWallet, token) => [
+        // Use similar arguments to the publicSale (though not necessarily).
+        // See also presale constant on
+        // migrations/3_deploy_public_sale_contracts.js file.
+        fundWallet,  // wallet
+        token.address,  // token contract
+        65000,  // rate
+        web3.toWei(5000410, "finney"),  // cap
+        web3.toWei(99, "finney"),  // individualMinPurchaseWei
+        [
+            currentTimestamp - 7 * day,  // a week ago
+            web3.toWei(5, "ether"),
+            currentTimestamp + 7 * day,  // a week later
+            web3.toWei(10, "ether"),
+            currentTimestamp + 14 * day,  // closingTime: 2 weeks later
+            0,
+        ],
+    ],
+    "CarryPublicTokenCrowdsale (opened; phase 2)": (fundWallet, token) => [
+        // Use similar arguments to the publicSale (though not necessarily).
+        // See also presale constant on
+        // migrations/3_deploy_public_sale_contracts.js file.
+        fundWallet,  // wallet
+        token.address,  // token contract
+        65000,  // rate
+        web3.toWei(5000410, "finney"),  // cap
+        web3.toWei(99, "finney"),  // individualMinPurchaseWei
+        [
+            currentTimestamp - 14 * day,  // 2 weeks ago
+            web3.toWei(5, "ether"),
+            currentTimestamp - 7 * day,  // a week ago
+            web3.toWei(10, "ether"),
+            currentTimestamp + 14 * day,  // closingTime: 2 weeks later
+            0,
+        ],
+    ],
+    "CarryPublicTokenCrowdsale (already closed)": (fundWallet, token) => [
+        // Use similar arguments to the publicSale (though not necessarily).
+        // See also presale constant on
+        // migrations/3_deploy_public_sale_contracts.js file.
+        fundWallet,  // wallet
+        token.address,  // token contract
+        65000,  // rate
+        web3.toWei(5000410, "finney"),  // cap
+        web3.toWei(99, "finney"),  // individualMinPurchaseWei
+        [
+            currentTimestamp - 31 * day,  // a month ago
+            web3.toWei(5, "ether"),
+            currentTimestamp - 14 * day,  // 2 weeks ago
+            web3.toWei(10, "ether"),
+            currentTimestamp - 7 * day,  // closingTime: a week ago
```

Comment:
Changing this line to `currentTimestamp + 7 * day` somehow passes the test? Can you check if you can reproduce this?

Reply:
ok looks like the test `"should not receive ethers if it's not opened yet or already closed"` always succeeds because it tries to send only 5 finney.

Reply:
Addressed by 6cd719a61da0db30f418e2bc9da24acc1ca72faf.